driver-core/src/main/com/mongodb/internal/operation/BulkWriteBatch.java

```diff
@@ -282,21 +281,11 @@ private BulkWriteResult getBulkWriteResult(final BsonDocument result) {
     }
 
     private List<BulkWriteInsert> getInsertedItems(final BsonDocument result) {
-        if (payload.getPayloadType() == SplittablePayload.Type.INSERT) {
-
-            Stream<WriteRequestWithIndex> writeRequests = payload.getWriteRequestWithIndexes().stream();
-            List<Integer> writeErrors = getWriteErrors(result).stream().map(BulkWriteError::getIndex).collect(Collectors.toList());
-            if (!writeErrors.isEmpty()) {
-                writeRequests = writeRequests.filter(wr -> !writeErrors.contains(wr.getIndex()));
-            }
-            if (payload.getPosition() < payload.size()) {
-                writeRequests = writeRequests.filter(wr -> wr.getIndex() < payload.getPosition());
-            }
-            return writeRequests
-                    .map(wr -> new BulkWriteInsert(wr.getIndex(), payload.getInsertedIds().get(wr.getIndex())))
-                    .collect(Collectors.toList());
-        }
-        return Collections.emptyList();
+        Set<Integer> writeErrors = getWriteErrors(result).stream().map(BulkWriteError::getIndex).collect(Collectors.toSet());
+        return payload.getInsertedIds().entrySet().stream()
```

Comment:
It's safe to use SplittablePayload`s insertedIds map directly, since it will only contain the ids from the previously inserted batch.

Reply:
So much neater and easier to read.